Vagrantfile

```diff
@@ -1,26 +1,205 @@
+PODMAN_E2E_PODMAN_BRANCH = "main"
+PODMAN_E2E_SKOPEO_VERSION = "1.13.1"
+
 Vagrant.configure("2") do |config|
-    config.vm.box = "fedora/33-cloud-base"
-    config.vm.synced_folder '.', '/vagrant', disabled: true
+  config.vm.define "default" do |m|
+    m.vm.box = "bento/fedora-42"
+    m.vm.synced_folder '.', '/vagrant', disabled: true
 
-    config.vm.provider "virtualbox" do |v|
+    m.vm.provider "virtualbox" do |v|
       v.memory = 2048
       v.cpus = 2
     end
-    config.vm.provision "shell", inline: <<-SHELL
+    m.vm.provision "shell", inline: <<-SHELL
       set -e -u -o pipefail
       yum update -y
-      yum install -y git gcc docker wget pkg-config systemd-devel dbus-devel elfutils-libelf-devel libseccomp-devel clang-devel openssl-devel
+      yum install -y git gcc docker wget pkg-config systemd-devel dbus-devel elfutils-libelf-devel libseccomp-devel clang-devel openssl-devel just
       grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
       service docker start
     SHELL
 
-    config.vm.provision "shell", privileged: false, inline: <<-SHELL
+    m.vm.provision "shell", privileged: false, inline: <<-SHELL
       curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
       echo "export PATH=$PATH:$HOME/.cargo/bin" >> ~/.bashrc
 
       git clone https://github.com/containers/youki
     SHELL
   end
+
+  config.vm.define "rootful" do |m|
+    m.vm.box = "bento/fedora-42"
+    m.vm.synced_folder '.', '/vagrant', disabled: true
+
+    m.vm.provider "virtualbox" do |v|
+      v.memory = 2048
+      v.cpus = 2
+    end
+    m.vm.provision "shell", path: "./hack/set_root_login_for_vagrant.sh"
+    m.vm.provision "shell", inline: <<-SHELL
+      set -e -u -o pipefail
+      yum update -y
+      yum install -y git gcc docker wget pkg-config systemd-devel dbus-devel elfutils-libelf-devel libseccomp-devel clang-devel openssl-devel just
+      grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"
+      service docker start
+      curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+      echo "export PATH=$PATH:$HOME/.cargo/bin" >> ~/.bashrc
+    SHELL
+    m.ssh.username = 'root'
+    m.ssh.insert_key = 'true'
+  end
+
+  config.vm.define "containerd2youki" do |m|
+    m.vm.box = "bento/ubuntu-24.04"
+    m.vm.synced_folder '.', '/vagrant/youki', disabled: false
+
+    m.vm.provider "virtualbox" do |v|
+      v.memory = 4096
+      v.cpus = 4
+    end
+
+    m.vm.provision "bootstrap", type: "shell" do |s|
+      s.inline = <<-SHELL
+        set -e -u -o pipefail
+        export DEBIAN_FRONTEND=noninteractive
+        apt-get update && apt-get install -y \
+          make \
+          pkg-config         \
+          libsystemd-dev     \
+          libdbus-glib-1-dev \
+          build-essential    \
+          libelf-dev \
+          libseccomp-dev \
+          libbtrfs-dev \
+          btrfs-progs
+        
+        ARCH=$(uname -m)
```

Comment:
Add a process to get an architecture as a variable to support the ARM environment.